rudder-core/src/main/scala/com/normation/rudder/db/SlickSchema.scala

```diff
@@ -0,0 +1,676 @@
+  }
+
+
+  //////////
+
+  final case class Reports(
+      id                 : Option[Long]
+    , executionDate      : DateTime
+    , nodeId             : String
+    , directiveId        : String
+    , ruleId             : String
+    , serial             : Int
+    , component          : String
+    , keyValue           : String
+    , executionTimestamp : DateTime
+    , eventType          : String
+    , policy             : String
+    , msg                : String
+  )
+
+  class TableReports(tag: Tag) extends Table[Reports](tag, "ruddersysevents") {
+    def id                 = column[Long]("id", O.PrimaryKey, O.AutoInc) // This is the primary key column
+    def executionDate      = column[DateTime]("executiondate")
+    def nodeId             = column[String]("nodeid")
+    def directiveId        = column[String]("directiveid")
+    def ruleId             = column[String]("ruleid")
+    def serial             = column[Int]("serial")
+    def component          = column[String]("component")
+    def keyValue           = column[String]("keyvalue")
+    def executionTimeStamp = column[DateTime]("executiontimestamp")
+    def eventType          = column[String]("eventtype")
+    def policy             = column[String]("policy")
+    def msg                = column[String]("msg")
+
+    // Every table needs a * projection with the same type as the table's type parameter
+    def * = (
+        id.?, executionDate, nodeId, directiveId, ruleId, serial
+      , component, keyValue, executionTimeStamp, eventType, policy, msg
+    ) <> (Reports.tupled, Reports.unapply)
+  }
+
+
+  //////////
+
+  final case class GitCommitJoin (
+      gitCommit     : GitCommitId
+    , modificationId: ModificationId
+  )
+
+  class TableGitCommitJoin(tag: Tag) extends Table[GitCommitJoin](tag, "gitcommit") {
+    def gitCommit      = column[String]("gitcommit", O.PrimaryKey)
+    def modificationId = column[String]("modificationid")
+
+    //needed to map nested case classes
+    private[this] def toModel(t2: (String, String)) = GitCommitJoin(GitCommitId(t2._1), ModificationId(t2._2))
+    private[this] def toTuples(m: GitCommitJoin) = Some((m.gitCommit.value, m.modificationId.value))
+    private[this] val shape = (gitCommit, modificationId).shaped[(String, String)]
+    def * = shape <> (toModel, toTuples)
+  }
+
+
+  //////////
+
+  final case class RunProperties (
+      name : String
+    , value: String
+  )
+
+  class TableRunProperties(tag: Tag) extends Table[RunProperties](tag, "rudderproperties") {
+    def name  = column[String]("name", O.PrimaryKey)
+    def value = column[String]("value")
+
+    def * = (name, value) <> (RunProperties.tupled, RunProperties.unapply)
+  }
+
+  //////////
+
+  final case class AgentRun(
+      nodeId      : String
+    , date        : DateTime
+    , nodeConfigId: Option[String]
+    , isCompleted : Boolean
+    , insertionId : Long // PURPOSE ?
```

Comment:
the goal is to have the id of the reports - rationnal is: latest we receive is the latest sent (network paquet ordering is more reliable than nodes times)